### PR TITLE
issue: 1067880 Scatter/gather array helper

### PR DESCRIPTION
**Pull request by olgk**
sg_array allows to represent scatter/gather buffers array as virtually
contiguous one-dimensional vector/array of one buffer.

get_data() requests some amount of data and returns pointer to it.
When there is no requested amount of data get_data() will return
remaining from current SGE and next call will move to next SGE or
return NULL.

Signed-off-by: Oleg Kuporosov <olegk@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1413/ for details (Mellanox internal link).

File: src/vma/util/sg_array.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+#ifndef SG_ARRAY_H
+#define SG_ARRAY_H
+
+#include <stdio.h>
+#include "util/verbs_extra.h"
+
+//@@ sg_array - helper class on top of scatter/gather elements array.
+//Represent it like a virtual one dimension vector/array.
+
```
Comment (igor-ivanov) [inline src/vma/util/sg_array.h]: please consider using traditional comment style as
`/**`
` * wakeup class that adds a wakeup functionality to socket (tcp and udp) and epoll using a pipe.`
` */`

File: tests/gtest/aux/sg_array.cc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	EXPECT_EQ(20,sa2.length());
+
+	sg_array	sa3(sge3,3);
+	EXPECT_EQ(3,sa3.get_num_sge());
+	EXPECT_EQ(30,sa3.length());
+
+}
+
+//@@ Tests for relative index
+//
```
Comment (igor-ivanov) [inline tests/gtest/aux/sg_array.cc]: please consider traditional comment style. See other gtest files.

File: tests/gtest/Makefile.am [hunk 3]
```diff
@@ -18,6 +18,8 @@ gtest_SOURCES = \
 	\
 	common/base.cc \
 	\
+	aux/sg_array.cc \
+	\
```
Comment (igor-ivanov) [inline tests/gtest/Makefile.am]: please add dummy aux_base.cc and aux_base.h

File: src/vma/util/sg_array.h (hunk 1 above)
Comment (olgk) [inline src/vma/util/sg_array.h]: addressed in commit of https://github.com/Mellanox/libvma/pull/418

File: tests/gtest/aux/sg_array.cc (hunk 2 above)
Comment (olgk) [inline tests/gtest/aux/sg_array.cc]: addressed in commit of https://github.com/Mellanox/libvma/pull/418

File: tests/gtest/Makefile.am (hunk 3 above)
Comment (olgk) [inline tests/gtest/Makefile.am]: addressed in commit of https://github.com/Mellanox/libvma/pull/418

**Comment by olgk** [comment]
This PR is closed by decision to have it as separate commit in https://github.com/Mellanox/libvma/pull/418

**Pull request closed without merging**